Frameworks/Foundation/NSProxy.mm

```diff
@@ -14,220 +14,317 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
 #import <Foundation/NSProxy.h>
+#import <Starboard/SmartTypes.h>
+
+#import <objc/objc-arc.h>
 
 @implementation NSProxy
+
+#pragma region NSProxy member functions
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (id)alloc {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self allocWithZone:nil];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (id)allocWithZone:(NSZone*)zone {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return class_createInstance(self, 0);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)dealloc {
-    UNIMPLEMENTED();
+    object_dispose(self);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)finalize {
-    UNIMPLEMENTED();
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)forwardInvocation:(NSInvocation*)anInvocation {
-    UNIMPLEMENTED();
+    [NSException raise:NSInvalidArgumentException
+                format:@"[NSProxy forwardInvocation:] was called directly. It must be overridden in a subclass."];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (NSMethodSignature*)methodSignatureForSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    [NSException raise:NSInvalidArgumentException
```

Comment:
nit: macro these, and use `%hs` + `__PRETTY_FUNCTION__` to get the method name in the macro